\begin{abstract}
	This thesis focuses on improving the extraction and hash encoding of spaced $k$-mers for bioinformatics applications. It explores the concept of spaced seeds, which improve similarity detection by allowing nonconsecutive matches within $k$-mers, albeit at the expense of increased computational complexity.
	
	The main goal of this research is to develop advanced software capable of rapid forward and reverse complement hashing for spaced $k$-mer in nucleotide sequences. This includes optimizing the hashing process to better handle large genomic datasets and minimize processing time and computational resources. The work includes the introduction of the DuoHash tool, an improved version of \ac{MISSH}, and we compare its performance with ntHash2. Results demonstrate how DuoHash performs on different da\-ta\-sets, showing its time efficiency and integrability with tools such as JellyFish. Finally, practical implications and suggestions for future research directions are discussed.
\end{abstract}